runtime/jvmti/jvmtiThread.c

```diff
@@ -886,7 +886,23 @@ jvmtiSetThreadLocalStorage(jvmtiEnv* env,
 
 		rc = getVMThread(currentThread, thread, &targetThread, TRUE, TRUE);
 		if (rc == JVMTI_ERROR_NONE) {
-			THREAD_DATA_FOR_VMTHREAD((J9JVMTIEnv *) env, targetThread)->tls = (void *) data;
+			J9JVMTIEnv *j9env = (J9JVMTIEnv *)env;
+			/* Use the current thread if the thread isn't specified. */
+			j9object_t threadObject = (NULL == thread) ? currentThread->threadObject : J9_JNI_UNWRAP_REFERENCE(thread);
+
+#if JAVA_SPEC_VERSION >= 19
+			rc = allocateTLS(vm, threadObject);
+			if (JVMTI_ERROR_NONE != rc) {
+				goto release;
+			}
+#endif /* JAVA_SPEC_VERSION >= 19 */
+			rc = createThreadData(j9env, targetThread, threadObject);
+			if (JVMTI_ERROR_NONE != rc) {
+				goto release;
+			}
+			jvmtiTLSGet(targetThread, threadObject, j9env->tlsKey)->tls = (void *)data;
```

Comment:
Do you need NULL check here ? The comment of `jvmtiTLSGet()` says it can return NULL on failure.

Reply:
Added fix and updated comments in latest commit

Reply:
Is it possible that NULL is set there by `jvmtiTLSSet()` ?

Reply:
In createThreadData we have
```
threadData = pool_newElement(j9env->threadDataPool);
if (NULL == threadData) {
	rc = JVMTI_ERROR_OUT_OF_MEMORY;
} else {
	jvmtiTLSSet(vmThread, thread, j9env->tlsKey, threadData)
}
```
So it cannot be set to null